dice_ml/explainer_interfaces/dice_KD.py

```diff
@@ -122,16 +121,35 @@ def _generate_counterfactuals(self, query_instance, total_CFs, desired_range=Non
                                                               verbose,
                                                               limit_steps_ls)
         self.cfs_preds = cfs_preds
-
+        # Decoding the output label back to original data type (e.g., str labels)
+        query_instance[self.data_interface.outcome_name] = \
+            self.decode_model_output(query_instance[self.data_interface.outcome_name])
+        if self.final_cfs_df is not None:
+            self.final_cfs_df[self.data_interface.outcome_name] = self.cfs_preds
+            self.final_cfs_df[self.data_interface.outcome_name] = \
+                self.decode_model_output(self.final_cfs_df[self.data_interface.outcome_name])
+            if self.final_cfs_df_sparse is not None:
+                self.final_cfs_df_sparse[self.data_interface.outcome_name] = \
```

Comment:
This looks like repeated code in dice_KD.py,  dice_genetic.py and dice_random.py. Could you create a common function instead for this code?

Reply:
Good idea. Done.

Reply:
@amit-sharma have you sent an update commit on this comment?

Reply:
pushed now.